src/nvim/tui/tui.c

```diff
@@ -166,14 +166,13 @@ void tui_start(TUIData **tui_p, int *width, int *height, char **term, bool *rgb)
   tui->seen_error_exit = 0;
   tui->loop = &main_loop;
   tui->url = -1;
-  // Because setting the default colors is delayed until after startup to avoid
-  // flickering with the default colorscheme background, any flush that happens
-  // during startup in turn would result in clearing invalidated regions with
-  // uninitialized attrs(black). Instead initialize clear_attrs with current
-  // terminal background so that it is at least not perceived as flickering, even
-  // though it may be different from the colorscheme that is set during startup.
-  tui->clear_attrs.rgb_bg_color = normal_bg;
-  tui->clear_attrs.cterm_bg_color = (int16_t)cterm_normal_bg_color;
+  // Setting the default colors is delayed until after startup to avoid flickering
+  // with the default colorscheme background. Consequently, any flush that happens
+  // during startup would result in clearing invalidated regions with zeroed
+  // clear_attrs, perceived as a black flicker. Initialize clear_attrs with -1
+  // instead to make clear_region()->update_attrs() use the current terminal background.
+  tui->clear_attrs.cterm_bg_color = -1;
```

Comment:
Should this be `0` instead? In `update_attrs` we have:

```c
    bg = (attrs.cterm_bg_color
          ? attrs.cterm_bg_color - 1 : (tui->clear_attrs.cterm_bg_color - 1));
    if (bg != -1) {
      UNIBI_SET_NUM_VAR(tui->params[0], bg);
      unibi_out(tui, unibi_set_a_background);
    }
```

But if `clear_attrs.cterm_bg_color` is `-1,` then `bg` ends up being `-2`, the `if` condition fails, and we still write a sequence to the terminal. It's a nonsense sequence, but it explains the `m` character (if I change `cterm_bg_color` to `0`, the test you wrote fails because the `m` is missing).

Reply:
Yeah this seemed suspect to me as well, and although what you say seems to make sense, my changes were result driven. If I set it to `0` I can still reproduce a black screen with `set notermguicolors` during startup. But I guess the nonsense sequence and `^m` result means we should for a different solution still...

Reply:
>If I set it to 0 I can still reproduce a black screen with set notermguicolors during startup.

Interesting, I cannot, and have never been able to, even before #28676. What are you doing to reproduce it? I am using

```vim
set notgc
colorscheme blue
au VimEnter * sleep 1
```

with `nvim --clean -u init.vim` and see no black screen (before and after #28676). If I remove `set notgc` then I see the black screen before #28676 but not after.

Reply:
Hmm just realized it may not be possible to reproduce on master. I first noticed it on my ext_ui branch and debugged it there. You can reproduce it by applying this diff:
```diff
diff --git a/runtime/lua/vim/_defaults.lua b/runtime/lua/vim/_defaults.lua
index 4684902410..fd47efbbe8 100644
--- a/runtime/lua/vim/_defaults.lua
+++ b/runtime/lua/vim/_defaults.lua
@@ -1,3 +1,4 @@
+vim.ui_attach(vim.api.nvim_create_namespace(''), { ext_cmdline = true }, function() end)
 --- Default user commands
 do
   vim.api.nvim_create_user_command('Inspect', function(cmd)
```

and running

    build/bin/nvim --clean +"se notermguicolors" +"sleep 2"

The timing of `nlua_init_defaults()`(which is earlier and where I run `vim.ui_attach` currently on that branch) vs startup cmd/scripts parsing seems significant here.

Not sure what that means, just noticed that `-1` gets rid of the black screen also in that case, but as it turns out for the wrong reasons. Still would like to look for a way that also prevents black screen in that case if possible.

Reply:
I still cannot reproduce the black screen with your patch using your command. The screen maintains my terminal's background color the whole time.

Reply:
Hmm no looking at `main.c` the fact that this is executed in `_defaults.lua` seems irrelevant. Should be able to reproduce this with `--cmd`/`+` as well.

Reply:
Still unsure about what was going on here but I changed the fix yet again to one that works in all cases and I hope makes sense.